Integrations/python/deephaven/KafkaTools/__init__.py

```diff
@@ -80,72 +89,163 @@ def _custom_avroSchemaToColumnDefinitions(*args):
-    ]
-
+        except Exception as e:
+            raise Exception(
+                "when of type dict, keyword argument 'offsets' has to map " +
+                "numeric partitions to either numeric offsets, or the constants DONT_SEEK and SEEK_TO_BEGINNING, " +
+                "instead got offsets=" + str(offsets)
+            ) from e
+    elif not isinstance(offsets, jpy.JType):
+        raise Exception(
+            "type " + type(offsets).__name__ +
+            "  of keyword argument 'offsets' not recognized; only str or dict allowed")
+
+    key = kwargs.pop('key', IGNORE)
+    value = kwargs.pop('value', IGNORE)
+    if key is IGNORE and value is IGNORE:
+        raise Exception(
+            "at least one keyword argument for specifying either a key or value is required; " + 
+            "they can't be both omitted, and they can't be both the IGNORE constant")
+
+    if len(kwargs) > 0:
+        raise Exception("excess keyword arguments not understood given: " + str(kwargs))
+
+    streaming_table = _java_type_.consumeToTable(consumer_props, topic, partitions, offsets, key, value)
 
-@_passThrough
-def consumeToTable(*args, **kwargs):
-    r = _commonKafkaArgs(args)
-    key_avro_schema = kwargs.pop('key_avro_schema', None)
-    value_avro_schema = kwargs.pop('value_avro_schema', None)
-    value_json = kwargs.pop('value_json', None)
-    if key_avro_schema is None and value_avro_schema is None and value_json is None:
-        # simple key/value case.
-        streaming_table = _java_type_.consumeToTable(r[0], r[1], r[2], r[3])
-    elif key_avro_schema is not None or value_avro_schema is not None:
-        if (value_json != None):
-            raise Exception('mixing json and avro not supported.')
-        mapping = getattr(_java_type_, 'DIRECT_MAPPING')
-        streaming_table = _java_type_.consumeToTable(r[0], r[1], r[2], r[3], key_avro_schema, mapping, value_avro_schema, mapping)
-    elif value_json is not None:
-        if not isinstance(value_json, list):
-            raise Exception('value_json keyword argument should be a list.')
-        col_def_list = _tuplesListToColDefsList(value_json)
-        streaming_table = _java_type_.consumeJsonToTable(r[0], r[1], r[2], r[3], col_def_list)
-    else:
-        raise Exception('Unknown keyword arguments: ' + str(kwargs))
     table_type = kwargs.pop('table_type', None)
     if table_type is None or table_type == 'append':
         return _stream_table_tools_.streamToAppendOnlyTable(streaming_table)
     elif table_type == 'streaming':
         return streaming_table
+    raise Exception("unknown value " + table_type + " for keyword argument 'table_type'")
+
+
+@_passThrough
+def avro(*args, **kwargs):
+    if len(args) < 1 and len(args) > 2:
+        raise Exception("one or two positional arguments expected, instead got " + len(args))
+    if _isStr(args[0]):
+        have_actual_schema = False
+        schema_name = args[0]
+        if len(args) < 2 or not _isStr(args[1]):
```

Comment:
That is interesting; I would have assumed there was a default.

Reply:
Since is a string I am kinda hesitant to guess "1", I've seen examples of things like '0.1', '1.1', '1', is kinda all over the place I think